core/client/user-manager.js

```diff
@@ -93,11 +93,9 @@ userManager = {
   },
 
   _checkForSkinUpdate(character, user, oldUser) {
-    const { guest } = user.profile;
-
     // check for skin updates
-    let hasSkinUpdate = !oldUser && !guest;
-    if (!hasSkinUpdate && !guest) {
+    let hasSkinUpdate = !oldUser && !user.username;
```

Comment:
I don't understand this change, is it for simplicity?

Reply:
Nop, now we have 3 state of user. 

The guest will have -> guest: true & username: true
The new created account but in onboarding -> guest: true & username: false
The normal account -> guest: false & username: false

In this case, it was for fixing a bug where character skin not appearing when user finishing is onboarding, needed to reload the page.

Reply:
I'm not a big fan of playing with the guest attribute on the user because it's used in several places and it seems sensitive. 

Would it be possible to base it on a session variable? For example make a `Session.set('onboarding', true)` when switching the account from ghost to classic user and then remove this variable at the end of the onboarding. We could also replay the onboarding if needed with this system.

For the skin bug I'll take care of it, it's on my to-do list 🙂